errors on tool

@claude i'm getting the following errors on the page when visiting:

https://manaandmeeples.co.nz/shop/
localhost:10000/api/games:1   Failed to load resource: net::ERR_CONNECTION_REFUSED
hook.js:608  Error: TypeError: Failed to fetch
    at TCGShop.jsx:200:11
    at TCGShop.jsx:232:5
    at gu (react-dom-client.production.js:8583:24)
    at ts (react-dom-client.production.js:10276:9)
    at ts (react-dom-client.production.js:10323:9)
    at ts (react-dom-client.production.js:10269:9)
    at ts (react-dom-client.production.js:10269:9)
    at es (react-dom-client.production.js:10221:13)
    at Zu (react-dom-client.production.js:10101:7)
    at es (react-dom-client.production.js:10241:7)
overrideMethod @ hook.js:608
localhost:10000/api/currency/detect:1   Failed to load resource: net::ERR_CONNECTION_REFUSED
localhost:10000/api/filters:1   Failed to load resource: net::ERR_CONNECTION_REFUSED
localhost:10000/api/filters/counts?language=English&sortBy=name&sortOrder=asc:1   Failed to load resource: net::ERR_CONNECTION_REFUSED
hook.js:608  Filter counts error: Failed to fetch

https://manaandmeeples.co.nz/shop/admin/login 

the page loads but I get this error when putting in my username and password:

POST http://localhost:10000/api/auth/admin/login net::ERR_CONNECTION_REFUSED
onSubmit @ AdminLogin.jsx:19
Yc @ react-dom-client.production.js:12317
(anonymous) @ react-dom-client.production.js:12867
It @ react-dom-client.production.js:1498
tf @ react-dom-client.production.js:12455
Ed @ react-dom-client.production.js:15306
kd @ react-dom-client.production.js:15274
AdminLogin.jsx:39  Login error: TypeError: Failed to fetch
    at onSubmit (AdminLogin.jsx:19:30)
    at Yc (react-dom-client.production.js:12317:13)
    at react-dom-client.production.js:12867:5
    at It (react-dom-client.production.js:1498:36)
    at tf (react-dom-client.production.js:12455:3)
    at Ed (react-dom-client.production.js:15306:7)
    at kd (react-dom-client.production.js:15274:7)

--- a/mana-meeples-shop/src/config/api.js
+++ b/mana-meeples-shop/src/config/api.js
@@ -5,7 +5,8 @@ const getApiUrl = () => {
   // Priority order:
   // 1. Explicit REACT_APP_API_URL from environment
   // 2. Auto-detect from current window location in production builds
-  // 3. Development fallback to localhost
+  // 3. Known production domain fallback
+  // 4. Development fallback to localhost
 
   if (process.env.REACT_APP_API_URL) {
     return process.env.REACT_APP_API_URL;
@@ -14,6 +15,12 @@ const getApiUrl = () => {
   // In production builds, try to detect the API URL from current location
   if (process.env.NODE_ENV === 'production' && typeof window !== 'undefined') {
     const { protocol, hostname, port } = window.location;
+
+    // Special handling for known production domain
+    if (hostname === 'manaandmeeples.co.nz' || hostname === 'www.manaandmeeples.co.nz') {
+      return 'https://manaandmeeples.co.nz/api';
+    }
+
     const portSuffix = port && port !== '80' && port !== '443' ? `:${port}` : '';
     return `${protocol}//${hostname}${portSuffix}/api`;
   }
@@ -27,7 +34,9 @@ export const API_URL = getApiUrl();
 // Export a function for components that need to check the URL dynamically
 export const getAPIUrl = () => API_URL;
 
-// For debugging - log the API URL in non-production environments
-if (process.env.NODE_ENV !== 'production') {
+// For debugging - always log the API URL to help troubleshoot connection issues
+if (typeof window !== 'undefined') {
   console.log('🔗 API URL configured as:', API_URL);
+  console.log('🌐 Current hostname:', window.location.hostname);
+  console.log('🏗️ Build environment:', process.env.NODE_ENV);
 }

